packages/react-label/src/components/Label/useLabel.tsx

```diff
@@ -20,15 +19,33 @@ const mergeProps = makeMergeProps<LabelState>({ deepMerge: labelShorthandProps }
  * @param props - props from this instance of Label
  * @param ref - reference to root HTMLElement of Label
  * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ * {@docCategory Label}
  */
 export const useLabel = (props: LabelProps, ref: React.Ref<HTMLElement>, defaultProps?: LabelProps): LabelState => {
   const state = mergeProps(
     {
       ref,
+      as: 'label',
+      size: 'medium',
+      required: {
+        as: 'span',
+      },
     },
-    defaultProps && resolveShorthandProps(defaultProps, labelShorthandProps),
-    resolveShorthandProps(props, labelShorthandProps),
+    defaultProps && resolveLabelShorthandProps(defaultProps),
+    resolveLabelShorthandProps(props),
   );
 
   return state;
 };
+
+/**
+ * Label will first need to check if required is a boolean or shorthandprops,
+ * this allows for the required prop handle both the default asterisk for required
+ * or a custom required text.
+ */
+const resolveLabelShorthandProps = (props: LabelProps) => {
+  const required = props.required === true ? '*' : props.required;
+  props = { ...props, required };
```

Comment:
I think it might be less expensive to just do:

```suggestion
  props.required = props.required === true ? '*' : props.required;
```